swan-lake/resources/featured-scenarios/build-a-change-data-capture-service-in-ballerina.md

```diff
@@ -0,0 +1,299 @@
+---
+layout: build-a-change-data-capture-service-left-nav-pages-swanlake
+title: Build a Change Data Capture (CDC) service in Ballerina 
+description: Capture and process database changes in real-time using Ballerina CDC connectors.
+keywords: ballerina, cdc, change data capture, database, streaming, API
+permalink: /learn/build-a-change-data-capture-service-in-ballerina/
+active: build-a-change-data-capture-service
+intro: In this guide, you’ll learn how to build a Change Data Capture (CDC) service using Ballerina to implement real-time fraud detection. The service monitors changes to a MySQL database and flags high-value transactions that exceed a defined threshold.
+---
+
+## Prerequisites
+
+To follow along with this tutorial, ensure you have:
+
+* [Ballerina 2201.12.0 (Swan Lake)](https://ballerina.io/downloads/) or later
+* A text editor  
+
+  > **Tip:** Preferably [Visual Studio Code](https://code.visualstudio.com/) with the [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.ballerina)
+* A terminal or command-line interface
+* MySQL 8.0 or later
+
+## Set up a MySQL server instance
+
+You can use one of the following methods to set up MySQL:
+
+1. **Install MySQL locally** from the [official MySQL site](https://dev.mysql.com/doc/mysql-getting-started/en/#mysql-getting-started-installing)
+2. **Use a server stack** like [XAMPP](https://www.apachefriends.org/) or [WampServer](https://wampserver.aviatechno.net/)
+3. **Run MySQL in Docker** using [these official instructions](https://dev.mysql.com/doc/mysql-installation-excerpt/8.0/en/docker-mysql-getting-started.html)
+4. **Use a cloud-hosted instance** from [Google Cloud SQL](https://cloud.google.com/sql), [Amazon RDS](https://aws.amazon.com/rds/mysql/), or [Azure Database for MySQL](https://azure.microsoft.com/en-us/services/mysql/)
+
+> **Tip:** Keep your MySQL connection details (host, port, username, and password) handy.
+
+## Set up the MySQL database
+
+### Enable CDC in MySQL
+
+CDC relies on MySQL’s binary logging feature, which is enabled by default in MySQL 8.0+ unless explicitly disabled.
+
+#### Verify binary logging
+
+Run the following SQL to confirm:
+
+```sql
+SHOW VARIABLES LIKE 'log_bin';
+```
+
+If `log_bin` is `ON`, you're good to go.
+
+#### Ensure proper CDC settings
+
+Also verify the following:
```

Comment:
```suggestion
Also, verify the following:
```

Reply:
Fixed in [098a221](https://github.com/ballerina-platform/ballerina-dev-website/pull/9891/commits/098a221ea98fef64881aca9b621b8c3733ff0ff3)